pkg/apis/common/validate.go

```diff
@@ -23,6 +23,10 @@ func ValidateTLSConfig(tlsConfig *TLSConfig) error {
 		clientKeySet = true
 	}
 
+	if !caCertSet && !clientCertSet && !clientKeySet && !tlsConfig.InsecureSkipVerify {
```

Comment:
Should validate the case that `InsecureSkipVerify` is true. Also missing a test case for that.

Reply:
added test for the case `InsecureSkipVerify` is true

Reply:
That is for when it's true. The false is insignificant so I'll remove the test case as well it's redundant.

Reply:
done